zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/DocumentenApiClient.kt

```diff
@@ -70,19 +86,27 @@ class DocumentenApiClient(
         authentication: DocumentenApiAuthentication,
         objectUrl: URI
     ): DocumentInformatieObject {
-        return checkNotNull(
-            webclientBuilder
-                .clone()
-                .filter(authentication)
-                .build()
-                .get()
-                .uri(objectUrl)
-                .retrieve()
-                .toEntity(DocumentInformatieObject::class.java)
-                .block()?.body
-        ) {
-            "Could not retrieve ${DocumentInformatieObject::class.simpleName} at $objectUrl"
+        val result = webclientBuilder
+            .clone()
+            .filter(authentication)
+            .build()
+            .get()
+            .uri(objectUrl)
+            .retrieve()
+            .toEntity(DocumentInformatieObject::class.java)
+            .block()?.body
+
+        if (result !== null) {
```

Comment:
This check is not necessary if you kept the throwing of the exception up top. So it would become something like:

```
        val result = checkNotNull(
            webclientBuilder
                .clone()
                .filter(authentication)
                .build()
                .get()
                .uri(objectUrl)
                .retrieve()
                .toEntity(DocumentInformatieObject::class.java)
                .block()?.body
        ) {
            "Could not retrieve ${DocumentInformatieObject::class.simpleName} at $objectUrl"
        }

        outboxService.send {
            DocumentInformatieObjectViewed(
                result.url.toString(),
                objectMapper.valueToTree(result)
            )
        }

        return result
```